### PR TITLE
Normaize URLs and test.

### PR DESCRIPTION
**Pull request by adamsilverstein**
## Summary

Addresses issue https://github.com/google/site-kit-wp/issues/1567

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: tests/phpunit/integration/Core/Util/Google_URL_Matcher_TraitTest.php [hunk 1]
```diff
@@ -205,6 +208,106 @@ public function data_is_domain_match() {
...
+	}
+
+	/**
+	 * @dataProvider data_url_normalizer
+	 *
+	 * @param string $domain   A domain.
+	 * @param string $compare  The domain or URL to compare.
+	 * @param bool   $expected Whether a match is expected or not.
+	 */
+	public function test_url_normalizer( $domain, $compare, $expected ) {
```
Comment (felixarntz) [inline tests/phpunit/integration/Core/Util/Google_URL_Matcher_TraitTest.php:226]: This entire `Google_URL_Normalizer`-specific test code should go into a separate test class, there should always be one test class per class to test. In other words this should go into `Google\Site_Kit\Tests\Core\Util\Google_URL_NormalizerTest`.

File: tests/phpunit/integration/Core/Util/Google_URL_Matcher_TraitTest.php [hunk 2]
```diff
@@ -205,6 +208,106 @@ public function data_is_domain_match() {
...
+			'equal URL except trailing slash'     => array(
+				'http://example.com',
+				'http://example.com/',
+				true,
+			),
+			'equal compare except trailing slash' => array(
+				'http://example.com/',
+				'http://example.com',
+				true,
+			),
```
Comment (felixarntz) [inline tests/phpunit/integration/Core/Util/Google_URL_Matcher_TraitTest.php:254]: See above, the trailing slash parts should be removed from here too.

**Review by felixarntz — CHANGES_REQUESTED** [review]
Looks solid for the most part, but there are a few things to fix.

**Review by felixarntz — APPROVED** [review]
@adamsilverstein I've quickly added the remaining code changes. Can you take care of QAing this, since you have most context and have some test sites? Would be great to provide some info and screenshots on your issue comment similar like Cole does usually.

**Comment by adamsilverstein** [comment]
Thanks for taking care of those final changes @felixarntz - I'll give this a test.

**Pull request merged**